core/src/main/scala/org/apache/spark/ErrorInfo.scala

```diff
@@ -58,7 +58,8 @@ private[spark] object SparkThrowableHelper {
   def getMessage(errorClass: String, messageParameters: Array[String]): String = {
     val errorInfo = errorClassToInfoMap.getOrElse(errorClass,
       throw new IllegalArgumentException(s"Cannot find error class '$errorClass'"))
-    String.format(errorInfo.messageFormat, messageParameters: _*)
+    String.format(errorInfo.messageFormat.replaceAll("<[a-zA-Z0-9_-]+>", "%s"),
```

Comment:
Shall we allow `a-zA-Z` only?

Reply:
I would be fine with that. I use camelCasing for parameters.
my only problem is that there is no compile time check. So people could use _ and -. And If they don't test the error message (well) they may not note that it got messed up.